ykrt/src/compile/jitc_yk/codegen/x64/mod.rs

```diff
@@ -1949,6 +1954,34 @@ impl<'a> Assemble<'a> {
         Ok(())
     }
 
+    fn cg_abs(&mut self, iidx: InstIdx, op: Operand) {
+        let bitw = op.bitw(self.m);
+        let [io_reg, tmp_reg] = self.ra.assign_gp_regs(
+            &mut self.asm,
+            iidx,
+            [
+                GPConstraint::Input {
+                    op: op.clone(),
+                    in_ext: RegExtension::ZeroExtended,
+                    force_reg: None,
+                    clobber_reg: false,
+                },
+                GPConstraint::Temporary,
+            ],
+        );
+        match bitw {
+            64 => {
+                dynasm!(self.asm
```

Comment:
I can't believe there's no `abs()` in x86!

I found [this implementation](https://stackoverflow.com/a/11927940/109414) which I find easier to understand:
```
mov ebx, eax
neg eax
cmovl eax, ebx ;if eax is now negative, restore its saved value
```

What do you think? People in the comments claiming this is what clang does and is fast.

Reply:
See what godbolt says clang does now? `cmovl` isn't obviously fast FWIW, so I'm mildly sceptical (but only mildly).